api/v2/checluster_types.go

```diff
@@ -119,6 +119,17 @@ type CheClusterDevEnvironments struct {
 	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
 	// +kubebuilder:validation:MaxLength=63
 	ServiceAccount string `json:"serviceAccount,omitempty"`
+	// Pod scheduler for the workspace pods.
+	// If not specified, the pod scheduler is set to the default scheduler on the cluster.
+	// +optional
+	PodSchedulerName string `json:"podSchedulerName,omitempty"`
+	// StartTimeout determines the maximum duration a workspace can take to start
+	// before it is automatically failed.
+	// Duration should be specified in a format parseable by Go's time package, e.g.
+	// "15m", "20s", "1h30m", etc.
+	// +optional
+	// +kubebuilder:default:="5m"
+	StartTimeout string `json:"startTimeout,omitempty"`
```

Comment:
I am wondering why not put this into seconds, like this?
`StartTimeoutSeconds *int32 `json:"startTimeoutSeconds,omitempty"``
The reasons are the following:
- simplify validation on checluster level
- dashboard shouldn't parse the value but always uses timeout in seconds

Reply:
I chose `string` to match the expected input that the [DevWorkspaceOperatorConfig expects](https://github.com/devfile/devworkspace-operator/blob/main/apis/controller/v1alpha1/devworkspaceoperatorconfig_types.go#L131).

However, since the progressTimeout field of the DWOC supports seconds, I could simply convert the int32 to string and append an `s`.  The maximum `int32` value seems to be `2147483647` which is ~596523 hours, so there doesn't seem to be any issue about not being able to set the start timeout high enough. 

I'll make your suggested modification :+1: